frontend/src/utilities/time.ts

```diff
@@ -0,0 +1,23 @@
+export const timeDifference = (current: number, previous: number): string => {
+  const msPerMinute = 60 * 1000;
+  const msPerHour = msPerMinute * 60;
+  const msPerDay = msPerHour * 24;
+  const msPerMonth = msPerDay * 30;
+  const msPerYear = msPerDay * 365;
+
+  const elapsed = current - previous;
+
+  if (elapsed < msPerMinute) {
+    return Math.round(elapsed / 1000) + ' seconds ago';
+  } else if (elapsed < msPerHour) {
+    return Math.round(elapsed / msPerMinute) + ' minutes ago';
+  } else if (elapsed < msPerDay) {
+    return Math.round(elapsed / msPerHour) + ' hours ago';
+  } else if (elapsed < msPerMonth) {
+    return 'approximately ' + Math.round(elapsed / msPerDay) + ' days ago';
```

Comment:
Not sure if the `approximately` word is not redundant/too much for UX. I think stating `7 days ago` is already understood as an approximation. It's shorter and will fit on a single line easier than `approximately 7 days ago` as shown on the screenshots.

Reply:
https://www.patternfly.org/v4/ux-writing/numerics#absolute-or-relative-time